engine/schema/src/main/java/com/cloud/usage/dao/UsageNetworksDaoImpl.java

```diff
@@ -70,11 +69,10 @@ public void remove(long networkId, Date removed) {
             SearchCriteria<UsageNetworksVO> sc = this.createSearchCriteria();
             sc.addAnd("networkId", SearchCriteria.Op.EQ, networkId);
             sc.addAnd("removed", SearchCriteria.Op.NULL);
-            UsageNetworksVO vo = findOneBy(sc);
-            if (vo != null) {
-                vo.setRemoved(removed);
-                vo.setState(Network.State.Destroy.name());
-                update(vo.getId(), vo);
+            List<UsageNetworksVO> usageNetworksVOs = listBy(sc);
+            for (UsageNetworksVO entry : usageNetworksVOs) {
+                entry.setRemoved(removed);
+                update(entry.getId(), entry);
```

Comment:
Shouldn't the state be set as well?

Reply:
No. It does not make sense to update the helper entry's state.

Suppose Usage runs at 00:15 and aggregates daily, and that the user deletes the network at 12:15.

If we update the entry's state to `Destroy`, on the next job execution, Usage would generate a usage record for the network usage between 00:15 - 12:15 with the state set to `Destroy` (implying that the network's state was `Destroy` between 00:15 - 12:15).

Then, Quota would inject `Destroy` as the network's state for that period into activation rules. This could result in the incorrect value being billed for tariffs based on the network's state.